storage/filestore/store.go

```diff
@@ -25,20 +26,21 @@ var (
 // Store implements a blob store
 type Store struct {
 	dir *Dir
+	log *zap.Logger
 }
 
 // New creates a new disk blob store in the specified directory
-func New(dir *Dir) *Store {
-	return &Store{dir}
+func New(dir *Dir, log *zap.Logger) *Store {
```

Comment:
That's a new one to me. I do not like that convention at all. In this case, `dir` is the chief thing that we are building a new `Store` object around, and `log` is just an ancillary note. Passing it first to the constructor seems backward.

Reply:
I would say to remove that from the convention or report exceptions or open an issue about it pointing to here.

Reply:
The reason log is in front is because human pattern recognition can eventually ignore `ctx, log` part in `Something(ctx, log, ...`... This of course would be better as the last argument, since then we stuff the non-important part as last, but this would end up as `Something(ctx, ..., log)`. And the convention for `ctx` is to be the first argument.

I.e. `ctx, log` keeps that non-important stuff together in case you have multiple them.

Reply:
I think my human pattern recognition may not work the same way, because I don't tend to look past `ctx, log, `. But more importantly, I don't want `log` to be ignored or overlooked, here, as it's being used as an element of a new object, not the typical use case of "here's a logger you can use during the execution of your function".

Reply:
Yeah, mostly the recognition is learned over time. Yeah, that's the same for ctx as well. To me the main argument is keeping `ctx, log` together. But also something I can survive without.... although silently I'm still annoyed by it :D.